src/Symfony/Bridge/Twig/Extension/SecurityExtension.php

```diff
@@ -24,37 +21,13 @@
  */
 class SecurityExtension extends AbstractExtension
 {
-    private $securityChecker;
-
-    public function __construct(AuthorizationCheckerInterface $securityChecker = null)
-    {
-        $this->securityChecker = $securityChecker;
-    }
-
-    public function isGranted($role, $object = null, $field = null)
```

Comment:
This is a BC break, it has to be deprecated first.

Reply:
Even though it's never called directly only through the twig function?

Reply:
@fabpot did the same thing without a BC break to HttpKernelExtension
812fbb444f1d1e4cafff5876ce7b22d5b8d4adf8

Reply:
Anyone extending this class (e.g. to replace the security extension by a custom one providing more functions) would be broken. IMO it's not acceptable.

Reply:
I can fix BC break if it really is considered a BC break. I don't want that to be a blocker.

Reply:
IMHO this is acceptable, for the reason stated: i.e. https://github.com/symfony/symfony/commit/812fbb444f1d1e4cafff5876ce7b22d5b8d4adf8 … namely, this is moving what always should have been to where it, um, should have been … calling it a BC break versus history is just bikeshedding

Reply:
Well, it was a BC break, we just accepted it (and merged as a feature, not a bugfix). 
There are a lot of classes in the codebase that should be used as extension points, and we don't change their public API from minor to minor, per semver and our BC promise. 
Let's see what others think